hera_qm/xrfi.py

```diff
@@ -1998,254 +2003,154 @@ def xrfi_run(ocalfits_files=None, acalfits_files=None, model_files=None,
-                                                                                                  run_check_acceptability=run_check_acceptability)
-    # Now that we've had a chance to load in all of the provided data products and
-    # run median filters when specified, we combine the metrics computed so far
-    # into a combien metrics object.
-    # it is possible that no metrics have been computed so far too.
-    if len(metrics) > 0:
-    # To do this, we use the metrics list, which stores all of the metrics that
-    # we calculated.
-        if len(metrics) > 1:
-            vdict['uvf_metrics'] = metrics[-1].combine_metrics(metrics[:-1],
-                                                method='quadmean', inplace=False)
-        else:
-            vdict['uvf_metrics'] = copy.deepcopy(metrics[-1])
-        vdict['uvf_metrics'].label = 'Combined metrics, round 1.'
-        alg_func = algorithm_dict['detrend_medfilt']
-        vdict['uvf_metrics'].metric_array[:, :, 0] = alg_func(vdict['uvf_metrics'].metric_array[:, :, 0],
-                                                     flags=~vdict['uvf_metrics'].weights_array[:, :, 0].astype(np.bool_),
-                                                     Kt=kt_size, Kf=kf_size)
-
-        # Flag on combined metrics
-        vdict['uvf_f'] = flag(vdict['uvf_metrics'], nsig_p=sig_init, run_check=run_check,
-                     check_extra=check_extra, run_check_acceptability=run_check_acceptability)
-        vdict['uvf_fws'] = watershed_flag(vdict['uvf_metrics'], vdict['uvf_f'], nsig_p=sig_adj, inplace=False,
-                                 run_check=run_check, check_extra=check_extra,
-                                 run_check_acceptability=run_check_acceptability)
-        vdict['uvf_fws'].label = 'Flags from combined metrics, round 1.'
-
-        flags += [vdict['uvf_fws']]
-        # OR everything together for initial flags
-        # for this, we use the flags list, which keeps
-        # all the flags we calculated in our first round of
-        # metrics/flagging.
-        vdict['uvf_init'] = copy.deepcopy(flags[0])
-        if len(flags) > 1:
-            for flg in flags[1:]:
-                vdict['uvf_init'] |= flg
-        vdict['uvf_init'].label = 'ORd flags, round 1.'
 
+    def _run_all_filters(median_round, 
+                         omnical_filter, omnical_chi2_filter, omnical_zscore_filter, 
+                         abscal_filter, abscal_chi2_filter, abscal_zscore_filter,
+                         omnivis_filter, cross_filter, auto_filter):
+        '''This function runs all possible filters, updating vdict as appropriate'''
+        
+        # Start with empty list of flags and metrics for later combination
+        metrics = []
+        flags = []
+        
+        # Modify these labels and parameters based on whether we're doing mean- or median-based statistics
+        label = {True: 'median', False: 'mean'}[median_round]
```

Comment:
note the True/False nature of median round in the docstring or use a stringe for median_round

Reply:
Done.